hdrline.c

```diff
@@ -461,6 +459,13 @@ void index_date(const struct ExpandoNode *node, void *data,
   ASSERT(len < sizeof(tmp2));
   mutt_strn_copy(tmp2, start, len, sizeof(tmp2));
 
+  // The sender's time zone might only be available as a numerical offset, so "%Z" behaves like "%z".
+  char *bigz = strstr(tmp2, "%Z");
```

Comment:
We should handle the case when multiple `%Z`s are present (to be fully compatible with the old parser)